source/elements/oneTBB/source/algorithms/blocked_ranges/blocked_rangeNd_cls.rst

```diff
@@ -0,0 +1,175 @@
+.. SPDX-FileCopyrightText: 2019-2024 Intel Corporation
+.. SPDX-FileCopyrightText: Contributors to the oneAPI Specification project.
+..
+.. SPDX-License-Identifier: CC-BY-4.0
+
+===============
+blocked_rangeNd
+===============
+**[algorithms.blocked_rangeNd]**
+
+Class template that represents a recursively divisible N-dimensional half-open interval.
+
+A ``blocked_rangeNd`` is the N-dimensional extension of ``blocked_range``, ``blocked_range2d``, and ``blocked_range3d``.
+You can interpret it as a Cartesian product of N instances of ``blocked_range``.
+Unlike ``blocked_range2d`` and ``blocked_range3d``, all ranges must be specified over the same ``Value`` type.
+Therefore, ``blocked_rangeNd<int,2>`` is a specialization of ``blocked_range2d`` where both dimensions are ``int``.
+
+.. code:: cpp
+
+    namespace oneapi {
+        namespace tbb {
+
+            template<typename Value, unsigned int N>
+            class blocked_rangeNd {
+            public:
+                // Types
+                using value_type = Value;
+                using dim_range_type = blocked_range<value_type>;
```

Comment:
One more naming question: consider `dimension_type` as the name, for clarity and also to avoid interpretation of "dim range type" as "shadowy" etc. Or does it seem important to highlight the range nature of the dimensions?

Reply:
Agreed. However, this name is consistent with the 2d and 3d range if I remember correctly. So if we change it here, we become inconsistent with the other types. This all would not have been an issue, but a previous comment of colleagues from Intel suggested that we make it public. And then inconsistencies matter. My opinion.

Reply:
Thanks for pointing this out, @tobiasweinzierl. I agree that consistency matters. Closing this question.